atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -446,16 +446,32 @@ private void refreshTokenRanges() {
                 }
             };
 
+    public <V, K extends Exception> V runWithRetry(
+            FunctionCheckedException<Cassandra.Client, V, K> fn,
+            boolean printHost) throws K {
+        return runWithRetryOnHost(getRandomGoodHost().getHost(), fn, printHost);
+    }
+
     public <V, K extends Exception> V runWithRetry(FunctionCheckedException<Cassandra.Client, V, K> fn) throws K {
-        return runWithRetryOnHost(getRandomGoodHost().getHost(), fn);
+        return runWithRetryOnHost(getRandomGoodHost().getHost(), fn, false);
     }
 
     public <V, K extends Exception> V runWithRetryOnHost(
             InetSocketAddress specifiedHost,
             FunctionCheckedException<Cassandra.Client, V, K> fn) throws K {
+        return runWithRetryOnHost(specifiedHost, fn, false);
+    }
+
+    private <V, K extends Exception> V runWithRetryOnHost(
+            InetSocketAddress specifiedHost,
+            FunctionCheckedException<Cassandra.Client, V, K> fn,
+            boolean printHost) throws K {
         int numTries = 0;
         boolean shouldRetryOnDifferentHost = false;
         while (true) {
+            if (printHost) {
```

Comment:
Can we always output this when running at trace level? If `specifiedHost.getHostName()` is slow, we can change the `if` above to be `if (log.isTraceEnabled())` instead.